File: examples/FaxSend.ts
import * as DropboxSign from "@dropbox/sign";
import * as fs from 'fs';

const faxApi = new DropboxSign.FaxApi();

// Configure HTTP basic authorization: api_key
faxApi.username = "YOUR_API_KEY";

// Upload a local file
const file = fs.createReadStream("example_signature_request.pdf");

// or, upload from buffer
const fileBuffer: DropboxSign.RequestDetailedFile = {
  value: fs.readFileSync("example_signature_request.pdf"),
  options: {
    filename: "example_signature_request.pdf",
    contentType: "application/pdf",
  },
};

// or, upload from buffer alternative
const fileBufferAlt: DropboxSign.RequestDetailedFile = {
  value: Buffer.from("abc-123"),
  options: {
    filename: "txt-sample.txt",
    contentType: "text/plain",
  },
};

const data: DropboxSign.FaxSendRequest = {
  files: [ file, fileBuffer, fileBufferAlt ],
  testMode: true,
  recipient: "16690000001",
  sender: "16690000000",
  coverPageTo: "Jill Fax",
  coverPageMessage: "I'm sending you a fax!",
  coverPageFrom: "Faxer Faxerson",
  title: "This is what the fax is about!",
};

const result = faxApi.faxSend(data);
result.then(response => {
  console.log(response.body);
}).catch(error => {
  console.log("Exception when calling Dropbox Sign API:");
  console.log(error.body);
});

File: examples/FaxSend.rb
require "dropbox-sign"

Dropbox::Sign.configure do |config|
  # Configure HTTP basic authorization: api_key
  config.username = "YOUR_API_KEY"
end

fax_api = Dropbox::Sign::FaxApi.new

data = Dropbox::Sign::FaxSendRequest.new
data.files = [File.new("example_signature_request.pdf", "r")]
data.test_mode = true
data.recipient = "16690000001"
data.sender = "16690000000"
data.cover_page_to = "Jill Fax"
data.cover_page_message = "I'm sending you a fax!"
data.cover_page_from = "Faxer Faxerson"
data.title = "This is what the fax is about!"

begin
  result = fax_api.fax_send(data)
  p result
rescue Dropbox::Sign::ApiError => e
  puts "Exception when calling Dropbox Sign API: #{e}"
end
